src/machine/machine_atsamd21.go

```diff
@@ -2082,3 +2083,29 @@ func setEPINTENSET(ep uint32, val uint8) {
 		return
 	}
 }
+
+// ResetProcessor should perform a system reset in preperation
+// to switch to the bootloader to flash new firmware.
+func ResetProcessor() {
+	arm.DisableInterrupts()
+
+	for !sam.NVMCTRL.INTFLAG.HasBits(sam.NVMCTRL_INTFLAG_READY) {
+	}
+
+	//NVMCTRL->STATUS.reg |= NVMCTRL_STATUS_MASK;
+	sam.NVMCTRL.STATUS.SetBits(0x111)
+
+	//NVMCTRL->ADDR.reg  = (uintptr_t)&NVM_MEMORY[APP_START / 4];
+	sam.NVMCTRL.ADDR.Set(uint32(uintptr(0x00002004))) // is this correct?
```

Comment:
This line right here is the one I am wanting to confirm, before I accidentally brick one of my boards...

Reply:
This is the line in the linker script that shows the starting address: https://github.com/tinygo-org/tinygo/blob/master/targets/atsamd21.ld#L4